fleet_adapter_mir/fleet_adapter_mir/robot_adapter_mir.py

```diff
@@ -0,0 +1,767 @@
+        docking_points,
+        mission_handle: MissionHandle
+    ):
+        if not ('description' in docking_points):
+            return
+        # Used for exiting while loop early in the event that for whatever
+        # reason the robot starts performing a different mission, while the
+        # original navigation mission is ongoing
+        docking_mission_identifier = mission_handle.execution.identifier
+        mission_queue_id = None
+        count = 0
+        retry_count = 10
+        while count < retry_count and not mission_queue_id:
+            if (docking_mission_identifier !=
+                    mission_handle.execution.identifier):
+                self.node.get_logger().info(
+                    f'[{self.name}] MissionHandle has changed to '
+                    f'{mission_handle.execution.identifier}, interrupting '
+                    f'navigation mission {docking_mission_identifier}. '
+                    f'Stopping original docking mission loop.')
+                break
+
+            try:
+                self.node.get_logger().info(
+                    f'Requested dock mission for [{self.name}]: '
+                    f'{docking_points}')
+                end_waypoint = docking_points['description']['end_waypoint']
+                if 'start_waypoint' in docking_points['description']:
+                    start_waypoint = \
+                        docking_points['description']['start_waypoint']
+                else:
+                    start_waypoint = None
+                mission_name = docking_points['mission_name']
+
+                mission_queue_id = self.api.dock(
+                    mission_name, start_waypoint, end_waypoint)
+                if mission_queue_id is None:
+                    self.node.get_logger().info(
+                        f'[{self.name}] Dock mission cannot be queued '
+                        f'(no mission queue id provided), retrying...')
+            except Exception as err:
+                self.node.get_logger().info(
+                    f'[{self.name}] Failed to request robot to dock to '
+                    f'destination: {err}. Retrying... ({count})'
+                )
+            count += 1
+            time.sleep(1)
+
+        if (mission_queue_id is None and
+                docking_mission_identifier !=
+                mission_handle.execution.identifier):
+            self.node.get_logger().info(
+                f'[{self.name}] Failed to request robot to dock to '
+                f'destination. Docking step {docking_mission_identifier} '
+                f'interrupted by new mission '
+                f'{mission_handle.execution.identifier}.'
+            )
+            return
+
+        if mission_queue_id is None:
+            self.node.get_logger().info(
+                f'[{self.name}] Failed to request robot to dock to '
+                f'destination. Maximum request navigate retries exceeded!')
+            mission_handle.execution.finished()
```

Comment:
I don't think we should trigger `finished` here... that would tell RMF to continue with a task even though the right conditions aren't met.

This seems like a case where we should use the `nav_issue_ticket` and maybe trigger a replan.